src/nupic/algorithms/ColumnPooler.cpp

```diff
@@ -274,6 +273,7 @@ class ColumnPooler // : public Serializable
     activateCells( cellOverlaps, predictedCells, active );
 
     // Learn
+    std::sort( active.getSparse().begin(), active.getSparse().end() );
```

Comment:
is there some benefit of the sort? does activateCells require sorted?

Reply:
The TemporalMemory (which is used to model the distal dendrites) *requires* that the active SDR be sorted.

Reply:
> [TM]  requires that the active SDR be sorted.

can we sort there? I think I saw it checks of sorted (O(n)) anyway, so delegating the sort to be done there might not hurt that much